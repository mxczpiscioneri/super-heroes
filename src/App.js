import React from 'react'
import { StatusBar } from 'react-native'
import Routes from './routes'
import Colors from './statics/colors'
import FavoriteContext from './components/Favorite/context'

const App = () => (
  <FavoriteContext.Provider>
    <StatusBar barStyle="light-content" backgroundColor={Colors.SECONDARY} />
    <Routes />
  </FavoriteContext.Provider>
)

export default App
