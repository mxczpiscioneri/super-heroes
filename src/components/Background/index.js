import React from 'react'
import PropTypes from 'prop-types'
import { ImageBackground } from './styles'
import { BackgroundImage } from '../../statics/images'

export const Background = props => (
  <ImageBackground source={BackgroundImage}>
    {props.children}
  </ImageBackground>
)

Background.propTypes = {
  children: PropTypes.any,
}

Background.defaultprops = {
  children: '',
}
