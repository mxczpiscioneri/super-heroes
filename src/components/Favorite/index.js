import React from 'react'
import PropTypes from 'prop-types'
import {
  Icon,
  Image,
} from './styles'
import {
  IconFavorite,
  IconFavoriteBorder,
} from '../../statics/images'

export const Favorite = props => {
  const { character, favoritesIds, setFavorite } = props
  const isFavorite = favoritesIds.indexOf(character.id) >= 0

  return (
    <Icon activeOpacity={0.6} onPress={() => setFavorite(character)}>
      <Image source={isFavorite ? IconFavorite : IconFavoriteBorder} />
    </Icon>
  )
}

Favorite.propTypes = {
  character: PropTypes.object.isRequired,
  favoritesIds: PropTypes.array.isRequired,
  setFavorite: PropTypes.func.isRequired,
}
