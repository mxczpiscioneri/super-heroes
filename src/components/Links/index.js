import React from 'react'
import PropTypes from 'prop-types'
import { Linking } from 'react-native'
import {
  Container,
  Link,
  Text,
} from './styles'

export const Links = props => (
  <Container>
    {props.urls.map(item => (
      <Link
        key={item.type}
        activeOpacity={0.8}
        onPress={() => { Linking.openURL(item.url) }}
      >
        <Text>{item.type}</Text>
      </Link>
    ))}
  </Container>
)

Links.propTypes = {
  urls: PropTypes.array.isRequired,
}
