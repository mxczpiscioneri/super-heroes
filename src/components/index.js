export * from './Background'
export * from './Character'
export * from './Favorite'
export * from './Links'
export * from './MoreInfo'
