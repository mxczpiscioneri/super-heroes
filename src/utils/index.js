export * from './Helpers'
export * from './Log'
