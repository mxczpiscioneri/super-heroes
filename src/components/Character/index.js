import React from 'react'
import PropTypes from 'prop-types'
import FavoriteContext from '../Favorite/context'
import {
  Container,
  Box,
  Image,
  Legend,
  FavoriteBox,
} from './styles'
import { Favorite } from '../Favorite'

export const Character = props => {
  const {
    character,
    image,
    onPress,
    legend,
  } = props

  return (
    <Container>
      <Box
        activeOpacity={0.9}
        onPress={onPress}
      >
        <Image source={{ uri: image }} />
        {legend &&
          <Legend>{character.name}</Legend>
        }
      </Box>
      <FavoriteBox>
        <FavoriteContext.Consumer>
          {context => (
            <Favorite {...context} character={character} />
          )}
        </FavoriteContext.Consumer>
      </FavoriteBox>
    </Container>
  )
}

Character.propTypes = {
  character: PropTypes.object.isRequired,
  image: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  legend: PropTypes.bool,
}

Character.defaultprops = {
  onPress: null,
  legend: '',
}
