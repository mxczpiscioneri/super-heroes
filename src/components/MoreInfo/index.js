import React from 'react'
import PropTypes from 'prop-types'
import Carousel from 'react-native-snap-carousel'
import { Dimensions } from 'react-native'
import { generateUrlImage } from '../../utils'
import {
  Container,
  Title,
  Box,
  Image,
  Legend,
} from './styles'

const { width } = Dimensions.get('window')

export const MoreInfo = props => (
  <Container>
    <Title>{props.title}</Title>

    <Carousel
      data={props.data}
      sliderWidth={width}
      itemWidth={166}
      renderItem={({ item }) => (
        <Box key={item.id}>
          {item.thumbnail &&
            <Image source={{ uri: generateUrlImage(item.thumbnail.path, item.thumbnail.extension, 'portrait_xlarge') }} />
          }
          <Legend>{item.title}</Legend>
        </Box>
      )}
    />
  </Container>
)

MoreInfo.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.array.isRequired,
}
