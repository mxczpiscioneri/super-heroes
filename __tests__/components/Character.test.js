
import React from 'react'
import renderer from 'react-test-renderer'
import FavoriteContext from '../../src/components/Favorite/context'
import { Character } from '../../src/components'
import { generateUrlImage } from '../../src/utils'
import { characterMock } from '../../__mocks__/characters'

it('Character snapshot', () => {
  const rendered = renderer.create(
    <FavoriteContext.Provider>
      <Character
        character={characterMock}
        legend
        image={generateUrlImage(characterMock.thumbnail.path, characterMock.thumbnail.extension, 'standard_fantastic')}
        onPress={() => null}
      />
    </FavoriteContext.Provider>
  ).toJSON()
  expect(rendered).toMatchSnapshot()
})
