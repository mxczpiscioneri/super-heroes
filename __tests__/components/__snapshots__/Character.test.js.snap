// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1A3555",
        "borderRadius": 6,
        "flexBasis": 0,
        "flexGrow": 1,
        "marginBottom": 8,
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 8,
        "opacity": 0.8,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://i.annihil.us/u/prod/marvel/i/mg/b/70/4c0035adc7d3a/standard_fantastic.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 6,
            "height": 200,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "marginBottom": 8,
            "marginTop": 16,
            "paddingHorizontal": 8,
          },
        ]
      }
    >
      Ajaxis
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "right": 16,
          "top": 16,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/statics/images/icon_favorite_border.png",
          }
        }
        style={
          Array [
            Object {},
          ]
        }
      />
    </View>
  </View>
</View>
`;
